File: src/runtime/boxes/fence/bb_fence.java
package bb;

/**
 * bb_fence.java — FENCE: succeed once; β cuts (no retry)
 * Port of bb_fence.c / bb_fence.s
 *
 *   FENCE_α:  fired=1;            goto FENCE_γ;
 *   FENCE_β:                      goto FENCE_ω;
 *   FENCE_γ:  return spec(Σ+Δ,0);
 *   FENCE_ω:  return spec_empty;
 */
public class bb_fence extends bb_box {

    public bb_fence(MatchState ms) { super(ms); }

    @Override public Spec α() { return new Spec(ms.delta, 0); }  // FENCE_γ

    @Override public Spec β()  { return null; }                   // FENCE_ω
}

File: src/runtime/boxes/fence/bb_fence.cs
// bb_fence.cs — FENCE: succeed once; β cuts (no retry)
// Mirrors src/runtime/boxes/fence/bb_fence.c
//
// α: γ zero-width (always succeeds first time)
// β: ω (cut — no retry)

namespace Snobol4.Runtime.Boxes;

public sealed class bb_fence : IByrdBox
{
    public Spec Alpha(MatchState ms) => Spec.ZeroWidth(ms.Cursor);
    public Spec Beta(MatchState ms)  => Spec.Fail;
}
